src/ratchet.rs

```diff
@@ -146,14 +160,32 @@ impl Ratchet {
     /// let mut ratchet = Ratchet::new(&mut rand::thread_rng());
     /// ratchet.inc_by(3);
     ///
-    /// println!("{:?}", String::from(&ratchet));
+    /// println!("{:?}", ratchet);
     /// ```
     pub fn inc_by(&mut self, n: usize) {
-        let (jumped, _) = inc_by(self, n);
-        *self = jumped;
+        if n == 0 {
+            return;
+        } else if n >= LARGE_EPOCH_LENGTH - self.combined_counter() {
+            // `n` is larger than at least one large epoch jump
+            let jump_count = self.next_large_epoch();
+            self.inc_by(n - jump_count);
+            return;
+        } else if n >= MEDIUM_EPOCH_LENGTH - self.small_counter as usize {
+            // `n` is larger than at least one medium epoch jump
+            let jump_count = self.next_medium_epoch();
+            self.inc_by(n - jump_count);
+            return;
+        }
+
+        self.small = Hash::from_chain([], self.small, n);
+        self.small_counter += n as u8;
     }
```

Comment:
I inlined this function and made it mutable (& made `next_large_epoch` and `next_medium_epoch` mutable, too).

Seems like more canonical rust style.